tests/cpp_integration_tests/test_main_model.cpp

```diff
@@ -1285,94 +1285,121 @@ TEST_CASE("Test main model - runtime dispatch") {
+                sym_output_from_row_based.add_buffer("node", node_output_from_row_based.size(),
+                                                     node_output_from_row_based.size(), nullptr,
+                                                     node_output_from_row_based.data());
+                MutableDataset sym_output_from_columnar{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
+                sym_output_from_columnar.add_buffer("node", node_output_from_columnar.size(),
+                                                    node_output_from_columnar.size(), nullptr,
+                                                    node_output_from_columnar.data());
+
+                base_model.calculate(options, sym_output_from_base);
+                row_based_model.calculate(options, sym_output_from_row_based);
+                columnar_model.calculate(options, sym_output_from_columnar);
+
+                REQUIRE(node_output_from_columnar.size() == node_output_from_base.size());
+                REQUIRE(node_output_from_columnar.size() == node_output_from_row_based.size());
+
+                for (Idx idx = 0; idx < std::ssize(node_output_from_columnar); ++idx) {
+                    // check columnar updates work same way as row-based updates
+                    CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_row_based[idx].id));
+                    CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_row_based[idx].u_pu));
+                    // check update actually changed something
+                    CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_base[idx].id));
+                    if (idx == 0) { // sym_load node
+                        CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_base[idx].u_pu));
+                    } else {
+                        CHECK(node_output_from_columnar[idx].u_pu != doctest::Approx(node_output_from_base[idx].u_pu));
+                    }
+                }
+            }
 
-            for (Idx idx = 0; idx < std::ssize(node_output_from_columnar); ++idx) {
-                // check columnar updates work same way as row-based updates
-                CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_row_based[idx].id));
-                CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_row_based[idx].u_pu));
-                // check update actually changed something
-                CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_base[idx].id));
-                if (idx == 0) { // sym_load node
-                    CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_base[idx].u_pu));
-                } else {
-                    CHECK(node_output_from_columnar[idx].u_pu != doctest::Approx(node_output_from_base[idx].u_pu));
+            SUBCASE("Without IDs") {
```

Comment:
Addressing your comment about extensive tests: I don't think validation tests need to be added to test this. If it works, then the results of a specific test should be the same (validation doesn't change) and this already checks if it works.